packages/mom5/package.py

```diff
@@ -0,0 +1,317 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# Copyright 2023 ACCESS-NRI
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import install, join_path, mkdirp
+
+# https://spack.readthedocs.io/en/latest/build_systems/makefilepackage.html
+class Mom5(MakefilePackage):
+    """MOM is a numerical ocean model based on the hydrostatic primitive equations."""
+
+    homepage = "https://www.access-nri.org.au"
+    git = "https://github.com/ACCESS-NRI/mom5.git"
+
+    maintainers = ["harshula"]
+
+    version("spack-build", branch="spack-build")
+
+    # Depend on virtual package "mpi".
+    depends_on("mpi")
+    depends_on("oasis3-mct")
+    depends_on("datetime-fortran")
+    depends_on("netcdf-fortran@4.5.2:")
+    depends_on("netcdf-c@4.7.4:")
+    # TODO: For initial verification we are going to use static pio.
+    #       Eventually we plan to move to shared pio
+    # ~shared requires: https://github.com/spack/spack/pull/34837
+    depends_on("parallelio~pnetcdf~timing~shared")
+    depends_on("libaccessom2")
+    depends_on("hdf5")
```

Comment:
`cice5` doesn't have an explicit dependency on `hdf5` so I'm wondering if this is necessary?

Reply:
1) In case you missed this:
```
MOM5.git/bin/environs.nci.gfortran
3:         module load hdf5
```
2) Since `netcdf-c` depends on `hdf5`, I'm happy to remove it, if you want.
3) Perhaps you could update `environs.nci.gfortran` too?

Reply:
I agree that it seems inconsistent with the `environs.nci.gfortran` but we **know** the spack `netcdf-c` package depends on it already, and in a non-straightforward way, so best to leave `netcdf-c` to specify that

https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/netcdf-c/package.py#L128-L142

Removing it from `environs.nci.gfortran` should be a separate issue.